code/weapon/weapon.h

```diff
@@ -330,8 +337,10 @@ struct weapon_info
 	int max_seeking;						// how many seekers can be active at a time if multilock is enabled. A value of one will lock stuff up one by one.
 	int max_seekers_per_target;			// how many seekers can be attached to a target.
 
-	SCP_vector<int> ship_type_restrict;
+	SCP_vector<multilock_restriction> ship_restrict;
 	SCP_vector<SCP_string> ship_type_restrict_temp;
+	SCP_vector<SCP_string> ship_class_restrict_temp;
+	SCP_vector<SCP_string> ship_species_restrict_temp;
```

Comment:
Rather than using multiple vectors, why not have a single vector as follows:
`std::pair<MultilockRestrictionType, SCP_string> restriction_name;`
That will also be extensible if any additional types are added in the future.

Reply:
I had initially attempted this, but it doesn't work without modifications or possibly an overload of stuff_string_list. I can try it if you think it would be worth it.

Reply:
Please do.  You don't need to overload stuff_string_list; you can do something like this:
```
if (optional_string("+Ship Types:")) {
	SCP_vector<SCP_string> temp_names;
	stuff_string_list(temp_names);

	for (auto &name: temp_names)
		wip->restrict_names.emplace_back(MultilockRestrictionType::TYPE, name);
}
```